app/controllers/take_quiz_controller.rb

```diff
@@ -1,5 +1,13 @@
 class TakeQuizController < ApplicationController
   before_action :set_quiz
+  before_action :ensure_guest_or_signed_in, except: %i[guest_prompt set_guest_username]
+
+  def guest_prompt; end
+
+  def set_guest_username
+    session[:guest_username] = params[:guest_username]
+    redirect_to quiz_path(@quiz)
```

Comment:
It seems strange that this redirects you to the management aspect of the quiz (which tells you the answer)